observability-kit-starter/src/main/java/com/vaadin/observability/ObservabilityHandler.java

```diff
@@ -41,8 +41,13 @@ private static Logger getLogger() {
 
     private final String id = UUID.randomUUID().toString();
 
-    public Function<String, String> config = (key) -> null;
-    public BiConsumer<String, Map<String,Object>> exporter = (id, map) -> {};
+    public transient Function<String, String> config = (key) -> {
+        getLogger().error("Observability agent is not running");
+        return null;
+    };
+    public transient BiConsumer<String, Map<String,Object>> exporter = (id, map) -> {
+        getLogger().error("Observability agent is not running");
+    };
```

Comment:
Another though about serialization.
Since the fields are transient, we need to ensure they are instantiated again during deserialization, for example by implementing the `readObject(ObjectInputStream stream)` hook (e.g. as in VaadinSession).

An alternative could be to have them null by default and handle null values in the calling code.